src/core/symbology-ng/qgssymbolv2.h

```diff
@@ -280,10 +281,26 @@ class CORE_EXPORT QgsMarkerSymbolV2 : public QgsSymbolV2
     QgsMarkerSymbolV2( QgsSymbolLayerV2List layers = QgsSymbolLayerV2List() );
 
     void setAngle( double angle );
-    double angle();
+    double angle() const;
+
+    /** Set "en masse" expression for angle
+     */
+    void setAngleExpression( const QString & exprStr );
+    /** Get "en masse" expression for angle
+     * @return empty if not an expession at the marker level
+     */
+    QString angleExpression() const;
 
     void setSize( double size );
-    double size();
+    double size() const;
+
+    /** Set "en masse" expression for size
+     */
+    void setSizeExpression( const QString & exprStr );
```

Comment:
these methods (setSizeExpression, sizeExpression, setWidthExpression, widthExpression) should accept/return a QgsDataDefined rather than QStrings. This will avoid the current situation where deactivating the size/width data defined button results in lose of the expression, and will future proof this work for when QgsDataDefined is enhanced in future (eg, there's plans for QgsDataDefined to allow for animation support, etc).